src/policy.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  Braid,
+  validateBIP32Path,
+  validateExtendedPublicKey,
+  validateRootFingerprint,
+} from "unchained-bitcoin";
+import { WalletPolicy } from "./vendor/ledger-bitcoin";
+import { BitcoinNetwork } from "./types";
+
+export class KeyOrigin {
+  xfp: string;
+  bip32Path: string;
+  xpub: string;
+  network: BitcoinNetwork;
+
+  constructor({ xfp, bip32Path, xpub, network }) {
+    this.network = network;
+    validateRootFingerprint(xfp);
+    this.xfp = xfp;
+    validateBIP32Path(bip32Path);
+    this.bip32Path = bip32Path;
+    validateExtendedPublicKey(xpub, network);
+    this.xpub = xpub;
+  }
+
+  /**
+   * Returns a key origin information in descriptor format
+   */
+  toString() {
+    let path = this.bip32Path;
+    if (this.bip32Path[0] === "m") {
+      path = this.bip32Path.slice(1);
+    }
+
+    return `[${this.xfp}${path}]${this.xpub}`;
+  }
+
+  // TODO: Needs a way to turn a serialized key origin to instance of class
+}
+
+export type MultisigScriptType = "sh" | "wsh" | "tr";
+
+/**
+ * Takes a Braid instance and translates it into a wallet policy template string
+ * @param {Braid} braid - instance of a braid object from unchained-bitcoin
+ * @returns valid policy template string
+ */
+export const getPolicyTemplateFromBraid = (braid: Braid) => {
+  let scriptType: MultisigScriptType | undefined;
+  let requiredSigners: number = Number(braid.requiredSigners);
+  let nested = false;
+  switch (braid.addressType) {
+    case "P2SH":
+      scriptType = "sh";
+      break;
+    case "P2WSH":
+      scriptType = "wsh";
+      break;
+    case "P2TR":
+      scriptType = "tr";
+      break;
+    case "P2SH-P2WSH":
+      scriptType = "wsh";
+      nested = true;
+      break;
+    default:
+      break;
+  }
+
+  if (!scriptType) {
+    throw new Error(`Unknown address type: ${braid.addressType}`);
+  }
+
+  const signersString = braid.extendedPublicKeys
+    .map((_, index) => `@${index}/**`)
+    .join(",");
+
+  // TODO: should this always assume sorted?
```

Comment:
imo yes

Reply:
Yeah. I think I agree but if we want this to be any sort of "reference" for policies then we might have to be more permissive.